### PR TITLE
remove keyboard ability to change value for readonly state in fast-number field

### PR DESCRIPTION
**Pull request by saicharan0124**
# Pull Request

## 📖 Description
removing the keyboard ability to change value for readonly state in fast-number field

### 🎫 Issues
#6712 

## 👩‍💻 Reviewer Notes

Go to the [number field demo](https://explore.fast.design/components/fast-number-field) or in Storybook and set the state to readonly.
Use the tab key to navigate to the input.
When focus is on the input, use the up or down arrow key to increment the value.
🤔 Expected Behavior
In readonly state, users shouldn't be able to change the value by using the up/down keys on keyboard.

😯 Current Behavior
In readonly state, users are able to change the value by using the up/down keys on keyboard.

## 📑 Test 
before bug fix :
[screen-capture (1).webm](https://github.com/microsoft/fast/assets/81006148/f8f6d60d-11dd-4292-b633-d8858cc93cbd)

After the fix :
[screen-capture (2).webm](https://github.com/microsoft/fast/assets/81006148/8b6c1cbb-6be2-4ee6-b72c-f93d8d4eeaae)

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [+] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [+] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [x] I have added a new component
- [+] I have modified an existing component
- [x] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [x] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by scomea** [comment]
You need to add a change file.  Go to repo root and 'yarn change'.  Choose "pre-release" in the options and upload the new commit.  Otherwise looks good.

**Review by scomea — APPROVED** [review]

**Comment by saicharan0124** [comment]
> 
@microsoft-github-policy-service agree

File: packages/web-components/fast-foundation/src/number-field/number-field.ts [hunk 1]
```diff
@@ -363,6 +363,10 @@ export class FASTNumberField extends FormAssociatedNumberField {
      * @internal
      */
     public handleKeyDown(e: KeyboardEvent): boolean {
+        
+        if (this.readOnly) {
+            return true;
+        }
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/number-field/number-field.ts:369]: We should likely check for disabled as well here.

```suggestion
        if (this.disabled || this.readOnly) {
            return true;
        }
```

File: change/@microsoft-fast-foundation-908e112e-07a6-4c72-b403-9f22841e6b86.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "remove keyboard ability to change value for readonly state in fast-number-field",
+  "packageName": "@microsoft/fast-foundation",
+  "email": "saicharan0124@gmail.com",
+  "dependentChangeType": "patch"
```
Comment (chrisdholt) [inline change/@microsoft-fast-foundation-908e112e-07a6-4c72-b403-9f22841e6b86.json:6]:
```suggestion
  "dependentChangeType": "prerelease"
```

File: packages/web-components/fast-foundation/src/number-field/number-field.ts [hunk 3]
```diff
@@ -363,6 +363,10 @@ export class FASTNumberField extends FormAssociatedNumberField {
      * @internal
      */
     public handleKeyDown(e: KeyboardEvent): boolean {
+        
+        if (this.disabled || this.readOnly) {
+            return;
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/number-field/number-field.ts:368]:
```suggestion
            return true;
```

File: packages/web-components/fast-foundation/src/number-field/number-field.ts [hunk 4]
```diff
@@ -363,6 +363,10 @@ export class FASTNumberField extends FormAssociatedNumberField {
      * @internal
      */
     public handleKeyDown(e: KeyboardEvent): boolean {
+        
+        if (this.disabled || this.readOnly) {
+            return true;
+        }
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/number-field/number-field.ts:369]: prettier...

```suggestion
        if (this.disabled || this.readOnly) {
            return true;
        }
```

**Comment by chrisdholt** [comment]
prettier is failing here but I'm going to merge this as it's been outstanding and fix that after the fact.

**Pull request merged**